en/performance/feature-tuning.html

```diff
@@ -77,25 +77,78 @@ <h2 id="when-to-use-fast-search">When to use fast-search for attribute fields</h
+<li><b>DAAT</b> <em>Document At A Time</em> scores documents completely one at a time. This requires multiple seeks in the term posting lists which is 
+ cpu cache unfriendly but allows non-linear ranking functions. </li>
+</ul>
 
-<p>The first-phase ranking score is assigned to the hit when it satisfies the query constraints and 
+Generally Vespa does <em>DAAT</em> (document-at-a-time) query evaluation and not <em>TAAT</em> (term-at-a time) for the reason listed above. 
+
+Ranking (score calculation) and matching (Does the document match the query logic) is not 
+fully two separate disjunct phases where one first find matches and in a later phase calculates the ranking score.
+Matching and <em>first-phase</em> score calculation is interleaved when using <em>DAAT</em>. 
+
+<p>The <em>first-phase</em> ranking score is assigned to the hit when it satisfies the query constraints and 
 at that point the term iterators are positioned at the document id and one can unpack additional data from the term posting lists, 
-for example for term proximity scoring used by <a href="../nativerank.html">nativeRank</a> ranking feature which also requires unpacking of positions of the term within the document.
+for example for term proximity scoring used by <a href="../nativerank.html">nativeRank</a> 
+ranking feature which also requires unpacking of positions of the term within the document.</p>
+
+<p>
+The way hybrid evaluation is done is that <em>TAAT</em> is used for sub branches 
+of the overall query tree which is not used for term based ranking. 
 
-Vespa can do a hybrid query evaluation combining TAAT and DAAT and using TAAT for sub-tree parts of the overall query tree which is not used for ranking, but used only for document filtering where 
-unpacking of posting lists is not needed. With <em>rank:filter</em> on a per field basis one can avoid unpacking on a per term/field basis while the TAAT evaluation works overall for the query sub tree. 
+Using <em>TAAT</em> can speed up query matching significantly (up to 20-30%) in cases where the query tree is large and complex 
+and where only parts of the query tree is used for term based ranking. Examples of branches that would require <em>DAAT</em> is using text ranking features 
+like <a href="../reference/rank-features.html">bm25/nativeRank</a>. The list of ranking features which can handle <em>TAAT</em> is long but  
+using <a href="../tensor-user-guide.html">attribute or tensor</a> features only can have the entire tree evaluated using <em>TAAT</em>.
+</p>
 
-Using TAAT can speed up query matching in some cases where the query tree is large and complex 
-and where only small parts of the query tree is used for ranking hits or ranking by a ranking expression only using <a href="../tensor-user-guide.html">attribute or tensor</a> features.
+<p>
+For example for a query where there is a user text query from an end user by one can use <em>userQuery()</em> YQL syntax 
```

Comment:
Note sure if I understood the sentence - is this better?
```suggestion
For example, for a query where there is a user text query from an end user, one can use <em>userQuery()</em> YQL syntax 
```